### PR TITLE
implement xiv get iscsi targets

### PR DESCRIPTION
**Pull request by oriyarde**
add xiv implementation for iscsi login

---
This change is [Reviewable](https://reviewable.io/reviews/ibm/ibm-block-csi-driver/136)

### CODE REVIEW

**Comment by igorgonibm** [comment]
___
*[controller/array_action/array_mediator_xiv.py, line 280 at r1](https://reviewable.io/reviews/ibm/ibm-block-csi-driver/136#-M316V0n3wCU-OrukpZe:-M316V0n3wCU-OrukpZf:b85nk32) ([raw file](https://github.com/ibm/ibm-block-csi-driver/blob/f04466661cef82113d1ec3ca6bacf29683bf38c4/controller/array_action/array_mediator_xiv.py#L280)):*
> ```Python
>             if interface.address:
>                 ips.append(interface.address)
>             if interface.address6:
> ```

why such join is needed?

**Comment by igorgonibm** [comment]
___
*[controller/array_action/array_mediator_xiv.py, line 280 at r1](https://reviewable.io/reviews/ibm/ibm-block-csi-driver/136#-M316V0n3wCU-OrukpZe:-M31AsTZ4KkdGcjQifDL:b-lvzhjt) ([raw file](https://github.com/ibm/ibm-block-csi-driver/blob/f04466661cef82113d1ec3ca6bacf29683bf38c4/controller/array_action/array_mediator_xiv.py#L280)):*
<details><summary><i>Previously, oriyarde wrote…</i></summary><blockquote>

same as in SVC, ipv6 should be in square brackets so when the csi node concatenate it to ":3260", the result is valid
</blockquote></details>

the concatination should be done in node

**Comment by igorgonibm** [comment]
___
*[controller/array_action/array_mediator_xiv.py, line 280 at r1](https://reviewable.io/reviews/ibm/ibm-block-csi-driver/136#-M316V0n3wCU-OrukpZe:-M31SzLJ2UqE3Trg66yV:b-tct0ti) ([raw file](https://github.com/ibm/ibm-block-csi-driver/blob/f04466661cef82113d1ec3ca6bacf29683bf38c4/controller/array_action/array_mediator_xiv.py#L280)):*
<details><summary><i>Previously, oriyarde wrote…</i></summary><blockquote>

node doesn't know which type of IP it is, and I don't think it should
</blockquote></details>

Can't node understand what ip type it is by its structure? 123.4 olr 11:22:33:44:55:66?
Some options here: https://stackoverflow.com/questions/22751035/golang-distinguish-ipv4-ipv6

**Comment by igorgonibm** [comment]
___
*[controller/array_action/array_mediator_xiv.py, line 280 at r1](https://reviewable.io/reviews/ibm/ibm-block-csi-driver/136#-M316V0n3wCU-OrukpZe:-M31kV_rAWodvsuriPmq:b901o7k) ([raw file](https://github.com/ibm/ibm-block-csi-driver/blob/f04466661cef82113d1ec3ca6bacf29683bf38c4/controller/array_action/array_mediator_xiv.py#L280)):*
<details><summary><i>Previously, oriyarde wrote…</i></summary><blockquote>

It just makes it complicated for the node, the controller has a guarantee of ip type, the node should not have to guess
</blockquote></details>

but this is ugly
for ip4 u return clean ip   but for ipv6 something weird  - ip in brackets.

**Comment by igorgonibm** [comment]
___
*[controller/array_action/array_mediator_xiv.py, line 280 at r1](https://reviewable.io/reviews/ibm/ibm-block-csi-driver/136#-M316V0n3wCU-OrukpZe:-M32AG9u15zPCuFIvByJ:bdau5wi) ([raw file](https://github.com/ibm/ibm-block-csi-driver/blob/f04466661cef82113d1ec3ca6bacf29683bf38c4/controller/array_action/array_mediator_xiv.py#L280)):*
<details><summary><i>Previously, oriyarde wrote…</i></summary><blockquote>

It is not less ugly than adding the brackets later in the node (it is even more efficient, since it saves the parsing check), and it is a common representation of a literal ipv6, so I don't find it weird or unclean.

there is also an option to remove the `+":3260"` logic in the node, since it is already the default port in iscsi-utils, and will probably never change.
</blockquote></details>

you mean [1:2:3:4:56:6] is valid ipv6?

File: controller/array_action/array_mediator_xiv.py [hunk 1]
```diff
@@ -270,10 +270,26 @@ def unmap_volume(self, volume_id, host_name):
             else:
                 raise controller_errors.UnMappingError(vol_name, host_name, ex)
 
-    def get_iscsi_targets_by_iqn(self):
+    def _get_iscsi_targets(self):
+        ip_interfaces = self.client.cmd.ipinterface_list().as_list
```
Comment (27149chen) [inline controller/array_action/array_mediator_xiv.py]: I guess the response of the command is iterable, you don't need as_list here.

Comment (oriyarde) [reply controller/array_action/array_mediator_xiv.py]: Done.

**Pull request merged**